pkg/domain/blueprintSpec_test.go

```diff
@@ -471,50 +483,98 @@ func TestBlueprintSpec_CheckEcosystemHealthAfterwards(t *testing.T) {
 }
 
 func TestBlueprintSpec_MarkInProgress(t *testing.T) {
-	//given
+	// given
 	spec := &BlueprintSpec{}
-	//when
+	// when
 	spec.MarkInProgress()
-	//then
+	// then
 	assert.Equal(t, spec, &BlueprintSpec{
 		Status: StatusPhaseInProgress,
 		Events: []Event{InProgressEvent{}},
 	})
 }
 
 func TestBlueprintSpec_MarkFailed(t *testing.T) {
-	//given
+	// given
 	spec := &BlueprintSpec{}
 	err := fmt.Errorf("test-error")
-	//when
+	// when
 	spec.MarkFailed(err)
-	//then
+	// then
 	assert.Equal(t, spec, &BlueprintSpec{
 		Status: StatusPhaseFailed,
 		Events: []Event{ExecutionFailedEvent{err: err}},
 	})
 }
 
 func TestBlueprintSpec_MarkBlueprintApplied(t *testing.T) {
-	//given
+	// given
 	spec := &BlueprintSpec{}
-	//when
+	// when
 	spec.MarkBlueprintApplied()
-	//then
+	// then
 	assert.Equal(t, spec, &BlueprintSpec{
 		Status: StatusPhaseBlueprintApplied,
 		Events: []Event{BlueprintAppliedEvent{}},
 	})
 }
 
 func TestBlueprintSpec_MarkCompleted(t *testing.T) {
-	//given
+	// given
 	spec := &BlueprintSpec{}
-	//when
+	// when
 	spec.MarkCompleted()
-	//then
+	// then
 	assert.Equal(t, spec, &BlueprintSpec{
 		Status: StatusPhaseCompleted,
 		Events: []Event{CompletedEvent{}},
 	})
 }
+
+func TestBlueprintSpec_ValidateDynamically(t *testing.T) {
+	type fields struct {
+		Id                   string
+		Blueprint            Blueprint
+		BlueprintMask        BlueprintMask
+		EffectiveBlueprint   EffectiveBlueprint
+		StateDiff            StateDiff
+		BlueprintUpgradePlan BlueprintUpgradePlan
+		Config               BlueprintConfiguration
+		Status               StatusPhase
+		PersistenceContext   map[string]interface{}
+		Events               []Event
+	}
+	type args struct {
+		possibleInvalidDependenciesError error
+	}
+	tests := []struct {
+		name           string
+		fields         fields
+		args           args
+		expectedPhase  StatusPhase
+		expectedEvents []Event
+	}{
+		{name: "statusphase invalid on error", fields: fields{}, args: args{possibleInvalidDependenciesError: assert.AnError}, expectedPhase: "invalid", expectedEvents: []Event{BlueprintSpecInvalidEvent{ValidationError: &InvalidBlueprintError{WrappedError: assert.AnError, Message: "blueprint spec is invalid"}}}},
+		{name: "statusphase valid on nil", fields: fields{}, args: args{possibleInvalidDependenciesError: nil}, expectedPhase: "validated", expectedEvents: []Event{BlueprintSpecValidatedEvent{}}},
```

Comment:
a few line breaks could increase readability.
Also, thanks a lot for increasing test coverage besides your own code!